src/robot/libdocpkg/xmlwriter.py

```diff
@@ -93,20 +93,18 @@ def _write_arguments(self, kw, writer):
                                  'repr': unicode(arg)})
             if arg.name:
                 writer.element('name', arg.name)
-            if arg.type is not arg.NOTSET:
-                writer.element('type', arg.type_repr)
+            if arg.types:
+                for type_repr in arg.types_reprs:
+                    writer.element('type', type_repr)
             if arg.default is not arg.NOTSET:
                 writer.element('default', arg.default_repr)
             writer.end('arg')
         writer.end('arguments')
 
     def _get_start_attrs(self, kw_type, kw, lib_source, writer):
-        if kw_type == 'init':
-            attrs = {}
-        else:
-            attrs = {'name': kw.name}
-            if kw.deprecated:
-                attrs['deprecated'] = 'true'
+        attrs = {'name': kw.name}
+        if kw_type != 'init' and kw.deprecated:
+            attrs['deprecated'] = 'true'
```

Comment:
I don't think there's need to check `kw_type` when deciding should `deprecated` attribute be added. Not doing that might remove the need to pass that info to this method altogether.